packages/wallet-sdk/src/sign/extension/ExtensionSigner.ts

```diff
@@ -28,12 +33,14 @@ export class ExtensionSigner implements Signer {
     extensionProvider.setAppInfo?.(appName, appLogoUrl, appChainIds);
     this.extensionProvider = extensionProvider;
 
-    this.extensionProvider.on('chainChanged', (chainId) => {
-      this.updateListener.onChainIdUpdate(Number(chainId));
-    });
-
-    this.extensionProvider.on('accountsChanged', (accounts) =>
-      this.updateListener.onAccountsUpdate(accounts as AddressString[])
+    const events: Parameters<ProviderEventCallback>[0][] = [
+      'connect',
+      'chainChanged',
+      'accountsChanged',
+      'disconnect',
+    ];
+    events.forEach((event) =>
+      this.extensionProvider.on(event, (data) => this.callback(event, data))
```

Comment:
🤮

Reply:
disconnect might be fired twice too 🤔  https://github.com/coinbase/coinbase-wallet-sdk/pull/1350/files#diff-bb3b1961b89bdeabb2d5abe40c8646da1dda20ca67ded4d932c32002e76d70daL105

Reply:
@nateReiners you are right.
but i think we are about to remove this ExtensionSigner soon.